packages/contracts/src/libraries/helpers/Validation.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.8.25;
+
+import "../../interfaces/ICrossL2Inbox.sol";
+import {Predeploys} from "../Predeploys.sol";
+import {ICrossL2Prover} from "../../interfaces/ICrossL2Prover.sol";
+
+contract Validation {
+    ICrossL2Prover private crossL2Prover;
+
+    enum ValidationMode {
+        CUSTOM,
+        CROSS_L2_INBOX,
+        CROSS_L2_PROVER_EVENT,
+        CROSS_L2_PROVER_RECEIPT
+    }
+
+    function _validate(
```

Comment:
make validate a callable external function .. or you know what can also be made a library , but then you have to pass the crossL2Prover along side `validate` function call..
let it be a contract , and make external calls each time to validate stuff